src/screens/SignUpScreen/index.js

```diff
@@ -0,0 +1,136 @@
+  inputRefs = {};
+
+  state = {
+    gender: undefined,
+    items: [
+      {
+        label: 'FEMALE',
+        value: 'female'
+      },
+      {
+        label: 'MALE',
+        value: 'male'
+      },
+      {
+        label: 'OTHER',
+        value: 'other'
+      }
+    ],
+    password: '',
+    email: '',
+    username: ''
+  };
+
+  signUpRequest = async () => {
+    const { username, email, gender, password } = this.state;
+    const { signUp, navigation } = this.props;
+    try {
+      await signUp({
+        username,
+        email,
+        gender,
+        password,
+        passwordConfirmation: password
+      });
+      navigation.navigate('Main');
+    } catch (err) {}
+  };
+
+  goToSignIn = () => {
+    navigation.dispatch(
+      StackActions.reset({
+        index: 0,
+        actions: [NavigationActions.navigate({ routeName: 'SignIn' })]
+      })
+    );
+  };
+
+  renderSpinner = () => {
+    if (this.props.loading) return <Spinner />;
+  };
+
+  render() {
+    const { username, email, gender, password, items } = this.state;
+    return (
+      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+        <View style={styles.blueCircle} />
+        <View style={styles.yellowCircle} />
+        <Text style={styles.titleText}>TARGET MVD</Text>
+        <View style={{ ...styles.formContainer, marginTop: 150 }}>
+          <Input
+            label="NAME"
+            value={username}
+            onChangeText={text => this.setState({ username: text })}
+          />
+          <Input
+            label="EMAIL"
+            value={email}
+            onChangeText={text => this.setState({ email: text })}
+          />
+          <Input
+            label="PASSWORD"
+            placeholder="MIN. 6 CHARS LONG"
+            secureTextEntry
+            value={password}
+            onChangeText={text => this.setState({ password: text })}
+          />
+          <Text style={{ ...styles.labelStyle, marginTop: 18 }}>GENDER</Text>
+          <RNPickerSelect
+            placeholder={{
+              label: 'SELECT YOUR GENDER...',
+              value: null
+            }}
+            items={items}
+            onValueChange={value => {
+              this.setState({
+                gender: value
+              });
+            }}
+            style={{ ...styles }}
+            value={gender}
+            ref={el => {
+              this.inputRefs.picker = el;
+            }}
+          />
+          <Text style={styles.errorMsg}>{this.props.error}</Text>
+          <BlackButton onPress={this.signUpRequest}>SIGN UP</BlackButton>
+          <SeparatorView />
+          <WhiteButton onPress={this.goToSignIn}>SIGN IN</WhiteButton>
+        </View>
+        {this.renderSpinner()}
+      </View>
+    );
+  }
+}
+
+SignUpScreen.propTypes = {
+  loading: bool.isRequired,
+  error: string.isRequired,
+  signUp: func.isRequired,
+  navigation: object.isRequired
+};
+
+const mapStateToProps = state => {
```

Comment:
You can simplify this a little bit by doing
```suggestion
const mapStateToProps = state => ({
  loading: state.auth.loading,
  error: state.auth.error,
});
```

The parentheses work like a return statement and are preferred for one-liners.